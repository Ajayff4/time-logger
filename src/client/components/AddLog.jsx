import React, { Component } from 'react'

class AddLog extends Component {
    tags = [
        "choose tag",
        "work",
        "reading",
        "excercise",
        "tutorials",
        "food",
        "fresh",
        "gaming",
        "cooking",
        "sleeping",
        "dish-washing",
        "cloth-washing",
        "bath"
    ]
    render() {
        return (
            <>
                <center>
                    <div id="add-log-form">
                        <div id="add-log-left">
                            <select>
                                {
                                    this.tags.map(tag => {
                                        return <option key={tag} value={tag}>{tag}</option>
                                    })
                                }
                            </select>
                            <br />
                            <button type="submit">Add</button>
                        </div >
                        <div id="add-log-right">
                            <textarea name="details" id="details" placeholder="Input details of log(optional)" />
                        </div>
                    </div>
                </center>
            </>
        )
    }
}

export default AddLog;