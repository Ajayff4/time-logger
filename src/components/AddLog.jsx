import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import axios from 'axios'
import { connect } from 'react-redux'
import * as actions from '../actions'
import PropTypes from 'prop-types'
/*
Have to work on log duration
*/

class AddLog extends Component {
    static propTypes = {
        username: PropTypes.string.isRequired,
        addLog: PropTypes.func.isRequired,
        failedToFetchData: PropTypes.func.isRequired
    }

    constructor(props) {
        super(props)
        this.state = {
            tag: "--",
            date: "",
            time: "",
            duration: 0,
            log_details: "",
            tags: []
        }
    }

    onChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    }

    onSubmit = (e) => {
        let curDate = new Date().toISOString().split('T')[0].toString();
        e.preventDefault()
        const {tag, duration, log_details } = this.state 
        //Adding new log
        axios.post('http://localhost:5000/rest/logs', {
            username: this.props.username,
            tag: tag,
            date: curDate,
            duration: duration,
            completed: false,
            log_details: log_details
        },{
            headers: {"Content-type": "application/json"}
        }).then(res => {
            this.props.addLog()
            if(res.data != null) {
                this.setState(this.initialState)
                alert("Log Saved Successfully")
            }
            this.props.history.push('/logs')
        }).catch(err => {
            this.props.failedToFetchData(`${err}`);
        })
        
    }

    componentDidMount() {
        //Getting all tags
        axios.get('http://localhost:5000/rest/tags')
        .then(res => {
            if(res.data != null) {
                let tagArray = []
                // eslint-disable-next-line
                res.data.map((tagCatObj, index) => {
                    for(let key in tagCatObj) {
                        if(key==="tag"){
                            tagArray.push(tagCatObj[key])
                        }
                    } 
                })
                this.setState({tags: [...tagArray]})
            }
        }).catch(err => {
            this.props.failedToFetchData(`${err}`);
        })


        if(this.props.editLogId) {
            this.handleEditButton(this.props.editLogId)
        }
    }

    createTag = () => {
        this.props.history.push("/add-tag")
    }

    render() {
        return (
            <>
                <center>
                    <h2>Add Log Component</h2>
                    <div id="form-css">
                        <form method="POST" onSubmit={this.onSubmit} >
                            <select
                                id="add-log-dropdown"
                                name="tag"
                                value={this.state.tag}
                                onChange={this.onChange}
                                required
                            >
                                <option key="choose" value="--">Choose tag</option>
                                {
                                    Object.entries(this.state.tags).map((tag, index) => {
                                        return <option key={tag[1]} value={tag[1]}>{tag[1]}</option>
                                    })
                                }
                            </select><br />
                            <textarea
                                name="log_details"
                                id="add-log-details"
                                placeholder="Input details of log(optional)"
                                value={this.state.log_details}
                                onChange={this.onChange}
                                required
                            /><br />
                            <button type="submit" id="add-log-submit">Add Log</button>
                            <br />
                            {/* eslint-disable-next-line */}
                            <a href="#" onClick={this.createTag}>Create a new tag</a>
                        </form>
                    </div>
                </center>
            </>
        )
    }
}

const mapStateToProps = state => {
    return {
        username: state.user.username
    }
}

const mapDispatchToProps = dispatch => {
    return {
        failedToFetchData: (error) => dispatch(actions.failedToFetchData(error)),
        addLog: () => dispatch(actions.addLog())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(AddLog))