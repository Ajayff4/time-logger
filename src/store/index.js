import { createStore, applyMiddleware } from 'redux'
import rootReducer from '../reducers/index'
//import logger from 'redux-logger'
import { composeWithDevTools } from 'redux-devtools-extension'
import rootSaga from '../saga'
import createSagaMiddleware from 'redux-saga'

const sagaMiddleware = createSagaMiddleware()
const store = createStore(
    rootReducer,
    composeWithDevTools(
        applyMiddleware(sagaMiddleware))
)

sagaMiddleware.run(rootSaga)

export default store