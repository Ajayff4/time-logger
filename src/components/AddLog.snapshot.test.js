import React from 'react'
import renderer from 'react-test-renderer'
import AddLog from './AddLog'
import { Provider } from 'react-redux';
import store from '../store';
import { BrowserRouter as Router } from 'react-router-dom';

describe('AddLog Snapshot Testing...', () => {
    it('renders component correctly', () => {
        const tree = renderer.create(<Provider store={store}><Router><AddLog /></Router></Provider> ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});