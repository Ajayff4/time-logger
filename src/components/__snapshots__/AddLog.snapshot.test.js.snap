// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddLog Snapshot Testing... renders component correctly 1`] = `
<div
  id="addLogComponent"
>
  <center>
    <h2>
      Add Log Component
    </h2>
    <div
      id="form-css"
    >
      <form
        method="POST"
        onSubmit={[Function]}
      >
        <select
          id="add-log-dropdown"
          name="tag"
          onChange={[Function]}
          required={true}
          value="--"
        >
          <option
            value="--"
          >
            Choose tag
          </option>
        </select>
        <br />
        <textarea
          id="add-log-details"
          name="log_details"
          onChange={[Function]}
          placeholder="Input details of log(optional)"
          required={true}
          value=""
        />
        <br />
        <button
          id="add-log-submit"
          type="submit"
        >
          Add Log
        </button>
        <br />
        <a
          href="#"
          onClick={[Function]}
        >
          Create a new tag
        </a>
      </form>
    </div>
  </center>
</div>
`;
